contracts/modules/PermissionManager/GeneralPermissionManager.sol

```diff
@@ -101,4 +99,115 @@ contract GeneralPermissionManager is IPermissionManager, Module {
         return allPermissions;
     }
 
+    /**
+    * @notice use to get all delegates
+    * @return address[]
+    */
+    function getAllDelegates() public view returns(address[]) {
+        return allDelegates;
+    }
+
+    /**
+    * @notice use to check if an address is a delegate or not
+    * @param _potentialDelegate the address of potential delegate
+    * @return bool
+    */
+    function isDelegate(address _potentialDelegate) public view returns(bool) {
+
+        require(_potentialDelegate != address(0));
+
+        if (delegateDetails[_potentialDelegate] != bytes32(0)) {
+            return true;
+        }else{
+            return false;
+        }
+    }
+
+    /**
+    * @notice Use to change one or more permissions for a single delegate at once
+    * @param _delegate Ethereum address of the delegate
+    * @param _multiModules Multiple module matching the multiperms, needs to be same length
+    * @param _multiPerms Multiple permission flag needs to be changed
+    * @return nothing
+    */
+    function changePermissionBulk(
+        address _delegate,
+        address[] _multiModules,
+        bytes32[] _multiPerms
+    )
+    external
+    withPerm(CHANGE_PERMISSION)
+    {
+        require(delegateDetails[_delegate] != bytes32(0), "Delegate details not set");
+        require(_multiModules.length != 0 && _multiPerms.length !=0);
+        
```

Comment:
add a check for equal length `require(_multiModules.length == _multiPerms.length, <Reason string>)`.